connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -603,9 +607,39 @@ public static ConfigInfos generateResult(String connType, Map<String, ConfigKey>
             }
             configInfoList.add(new ConfigInfo(configKeyInfo, configValueInfo));
         }
+
+        final List<ConfigValueInfo> nullConfigValues = buildConfigValueInfoWithNullValues(connectorProps);
+        nullConfigValues.forEach(configWithNullValue -> addError(configInfoList, configWithNullValue));
+        errorCount += nullConfigValues.size();
+
         return new ConfigInfos(connType, errorCount, groups, configInfoList);
     }
 
+    private static void addError(List<ConfigInfo> configInfoList, ConfigValueInfo configWithNullValue) {
+        final Optional<ConfigInfo> predefinedConfig = getConfigInfoForPredefinedConfig(configInfoList, configWithNullValue.name());
+        if (predefinedConfig.isPresent()) {
+            predefinedConfig.get()
+                    .configValue()
+                    .errors()
+                    .addAll(configWithNullValue.errors());
+        } else {
+            configInfoList.add(new ConfigInfo(null, configWithNullValue));
+        }
+    }
+
+    private static Optional<ConfigInfo> getConfigInfoForPredefinedConfig(List<ConfigInfo> configInfoList, String nullConfigName) {
+        return configInfoList.stream()
+                .filter(configInfo -> configInfo.configValue().name().equals(nullConfigName))
+                .findAny();
+    }
+
+    private static List<ConfigValueInfo> buildConfigValueInfoWithNullValues(Map<String, String> connectorProps) {
+        return connectorProps.entrySet().stream()
+                .filter(configEntry -> configEntry.getValue() == null)
+                .map(configEntry -> new ConfigValueInfo(configEntry.getKey(), String.format("Null Value Not Allowed for key %s.", configEntry.getKey())))
```

Comment:
I believe that including the name of the property in the error message is redundant as that information will be available already in the REST response.

I also think we may want to be clearer about the error message here. Users can't supply null values, but developers (by specifying `null` as the default value for a property in a `ConfigDef`, for example) definitely can, and we may want to make it clear which variety we're prohibiting.

What do you think about this?
```suggestion
                .map(configEntry -> new ConfigValueInfo(configEntry.getKey(), "The JSON literal `null` may not be used in connector configurations"))
```